### PR TITLE
[Feat] Add mcpack encode/decode protocol package

### PR DESCRIPTION
**Pull request by dongjiang1989**
### What type of PR is this?
/kind feature

- [x] Unittest
- [x] Coverage 80%+
- [x] Build Success

### What this PR does / why we need it:
 Add  mcpack encode/decode protocol package for serialization

### CODE REVIEW

**Review by zhouzhenling — APPROVED** [review]

File: pkg/codec/mcpack/decode.go [hunk 1]
```diff
@@ -0,0 +1,957 @@
...
+	"reflect"
+	"runtime"
+)
+
+var (
+	errEmptyKey      = errors.New("empty key")
+	errUnexpectedEnd = errors.New("unexpected end")
+)
+
+func Unmarshal(data []byte, v interface{}) error {
```
Comment (imlense) [inline pkg/codec/mcpack/decode.go:31]: Unmarshal add to codec interface

File: pkg/codec/mcpack/encode.go [hunk 2]
```diff
@@ -0,0 +1,832 @@
...
+	"fmt"
+	"reflect"
+	"runtime"
+	"sort"
+	"strings"
+	"sync"
+	"unicode"
+)
+
+func Marshal(v interface{}) ([]byte, error) {
```
Comment (imlense) [inline pkg/codec/mcpack/encode.go:29]: Marshal add to codec interface

File: pkg/codec/mcpack/decode.go (hunk 1 above)
Comment (dongjiang1989) [reply pkg/codec/mcpack/decode.go:31]: done

File: pkg/codec/mcpack/encode.go (hunk 2 above)
Comment (dongjiang1989) [reply pkg/codec/mcpack/encode.go:29]: done

**Pull request merged**